lib/kubernetes-deploy/kubernetes_resource.rb

```diff
@@ -192,5 +193,10 @@ def pretty_status
     def kubectl
       @kubectl ||= Kubectl.new(namespace: @namespace, context: @context, logger: @logger, log_failure_by_default: false)
     end
+
+    def template
```

Comment:
Memoizing _should_ be safe... From file, the result obviously won't change. Although resources do change cluster-side, the ones that don't have a file are child resources for which that shouldn't be true. For example, if the initial status check (before the deploy makes any changes) on a deployment resource memoized a template from the cluster, that'd be bad, since that template would change when the deploy proceeds. But the child resource objects are replaced by their parents every time the parent syncs.